libs/utils/trace.py

```diff
@@ -136,10 +136,14 @@ def __init__(self, platform, data_dir, events,
 
         # If we don't know the number of CPUs, check the trace for the
         # highest-numbered CPU that traced an event.
-        if 'cpus_count' not in self.platform:
+        if self.platform and 'cpus_count' not in self.platform:
             max_cpu = max(int(self.data_frame.trace_event(e)['__cpu'].max())
                           for e in self.available_events)
             self.platform['cpus_count'] = max_cpu + 1
+        # If a CPUs count is not available here, let's assume we are running on
+        # a unicore system and set cpus_count=1 so that the following analysis
+        # methods will not complain about the CPUs count not being available.
+        self.platform['cpus_count'] = self.platform.get('cpus_count', 1)
```

Comment:
Isn't that already handled by the above code ? In any case, this breaks `test_deriving_cpus_count`, aka the test that asserts the above code works.

Reply:
This should be fixed now...